### ISSUE TITLE
A password with special characters is not escaped right when create a docker-registry secret.

### ISSUE THREAD

**Issue by kamerblauwlicht** [kubernetes/kubectl#46]
**BUG REPORT**

**Kubernetes version** 

```
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.1", GitCommit:"1dc5c66f5dd61da08412a74221ecc79208c2165b", GitTreeState:"clean", BuildDate:"2017-07-14T02:00:46Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.1", GitCommit:"1dc5c66f5dd61da08412a74221ecc79208c2165b", GitTreeState:"clean", BuildDate:"2017-07-14T01:48:01Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: all cloudproviders
- **OS** : 16.04.3 LTS (Xenial Xerus) 
- **Kernel** : Linux master 4.4.0-87-generic #110-Ubuntu SMP Tue Jul 18 12:55:35 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**What happened**:

When you create a docker-registry secret with a special character you can't pull images. Looks like the special character is not escaped.

For example:

`kubectl create secret docker-registry index.docker.io-v1 --docker-server=https://index.docker.io/v1/ --docker-username=hello --docker-password=password$withspecialcharacter --namespace=test --docker-email=text@example.com`

You can't pull the image because the password is not correctly used. 

When you run this command it works:

`kubectl create secret docker-registry index.docker.io-v1 --docker-server=https://index.docker.io/v1/ --docker-username=hello --docker-password=password\$withspecialcharacter --namespace=test --docker-email=text@example.com`

`\$` instead of `$`

**What you expected to happen**:

I was expected that the special character can be used as a password without adding my own escape character.

**How to reproduce it** :

- Create an account for your private image repo (for example docker hub) 
- Use a special character in the password (for example: `mypassword$help`)
- Create the secret
- Try to deploy a pod with an image from docker hub
- Go get an error can't login to repo because of wrong username/password

**Anything else we need to know**:

Ask if you have any question ;-) Keep up the good work! You are all awesome.

**Comment by kamerblauwlicht** [comment]
oh deer. it's a shell command and special characters need to be escaped. 
ok continue and nevermind.

**Comment by josefkorbel** [comment]
Hello, I've came across this issue again and I dont think it is bash related. I've created a SO question. Here it is.

I was following [this][1] tutorial for deploying Django App to Kubernetes Cluster. I've created cloudsql credentials and exported them as in the tutorial

```
export DATABASE_USER=<your-database-user>
export DATABASE_PASSWORD=<your-database-password>
```
However my password was generated by LastPass and contains special characters, which are striped out in Kubernetes Pod thus making the password incorrect.

This is my password (altered, just showing the special chars)
`5bb4&sL!EB%e`

So i've tried various ways of exporting this string, echoing it out always show correct password, however in Kubernetes Dashboard the password is always incorrect (Also altered in DevTools, but some chars are just stripped out)

[![enter image description here][2]][2]

Things I've tried

```
export DATABASE_PASSWORD=$'5bb4&sL\!EB\%e'
export DATABASE_PASSWORD='5bb4&sL!EB%e'
```

Echoing is always good but kubernetes is always stripping it.

Deploying with `skaffold deploy`

  [1]: https://cloud.google.com/python/django/kubernetes-engine
  [2]: https://i.stack.imgur.com/yR5o9.png
